crates/sovereign-sdk/module-system/sov-modules-api/src/transaction.rs

```diff
@@ -71,6 +72,16 @@ impl<C: Context> Transaction<C> {
     }
 }
 
+impl<C: Context> TransactionTrait for Transaction<C> {
```

Comment:
I am not sure if I can calc hash this way or I need to add a hash field to Transaction struct

Reply:
depends on how many times we invoke the hash func.

if only once, we don't need it in the struct. else, it's better to have some form of calculate and then reuse the calculated value.